diff -burN Crypt-Random-1.23.orig/lib/Crypt/Random.pm Crypt-Random-1.23/lib/Crypt/Random.pm
--- Crypt-Random-1.23.orig/lib/Crypt/Random.pm	2004-11-01 22:51:54.875080848 +0100
+++ Crypt-Random-1.23/lib/Crypt/Random.pm	2004-11-01 22:55:04.162304816 +0100
@@ -14,7 +14,7 @@
 use Math::Pari qw(PARI floor Mod pari2pv pari2num lift); 
 use Carp; 
 use Data::Dumper;
-use Class::Loader;
+#use Class::Loader;
 use Crypt::Random::Generator;
 *import      = \&Exporter::import;
 
@@ -42,12 +42,13 @@
     local $| = 1;
 
     my $provider = _pickprovider(%params);
-    my $loader = new Class::Loader;
-    my $po = $loader->_load ( Module => "Crypt::Random::Provider::$provider", 
-                              Args => [ map { $_ => $params{$_} }
-                                qw(Strength Provider) ] )
-
-        or die "Unable to load module Crypt::Random::Provider::$provider - $!";
+#    my $loader = new Class::Loader;
+#    my $po = $loader->_load ( Module => "Crypt::Random::Provider::$provider", 
+#                              Args => [ map { $_ => $params{$_} }
+#                                qw(Strength Provider) ] )
+	eval "require Crypt::Random::Provider::$provider"
+        or die "Unable to load module Crypt::Random::Provider::$provider - $@";
+    my $po = "Crypt::Random::Provider::$provider"->new(%params);
     my $r = $po->get_data( %params );
 
     my $size     = $params{Size};
@@ -109,9 +110,12 @@
     $params{Verbosity} = 0 unless $params{Verbosity};
 
     my $provider = _pickprovider(%params); 
-    my $loader = new Class::Loader;
-    my $po = $loader->_load ( Module => "Crypt::Random::Provider::$provider", 
-                              Args => [ %params ] );
+#    my $loader = new Class::Loader;
+#    my $po = $loader->_load ( Module => "Crypt::Random::Provider::$provider", 
+#                              Args => [ %params ] );
+       eval "require Crypt::Random::Provider::$provider"
+        or die "Unable to load module Crypt::Random::Provider::$provider - $@";
+       my $po = "Crypt::Random::Provider::$provider"->new(%params);
     return $po->get_data( %params );
 
 
diff -burN Crypt-Random-1.23.orig/Makefile.PL Crypt-Random-1.23/Makefile.PL
--- Crypt-Random-1.23.orig/Makefile.PL	2004-11-01 22:51:54.877080544 +0100
+++ Crypt-Random-1.23/Makefile.PL	2004-11-01 22:55:14.223775240 +0100
@@ -7,6 +7,6 @@
     'VERSION_FROM' => 'lib/Crypt/Random.pm',
     'EXE_FILES'    => [ 'bin/makerandom' ],
     'PREREQ_PM'    => { 'Math::Pari' => '2.001804', 
-                        'Class::Loader' => '2.00' 
+#                        'Class::Loader' => '2.00' 
                       } 
 );
